pkg/operation/botanist/cleanup.go

```diff
@@ -72,7 +77,14 @@ func (b *Botanist) CleanKubernetesResources() error {
 		return err
 	}
 
+	apiGroupPathToBeChecked := make(map[string][]string)
+	//Avoid to change original data, we need a copy
 	for resource, apiGroupPath := range b.K8sShootClient.GetResourceAPIGroups() {
+		apiGroupPathToBeChecked[resource] = apiGroupPath
+	}
+	//For CSI managed PV, the deletion must take some while, need to be checked
+	apiGroupPathToBeChecked[kubernetesbase.PersistentVolumes] = []string{"api", "v1"}
```

Comment:
Why not adding this to https://github.com/gardener/gardener/blob/master/pkg/client/kubernetes/base/base.go#L51?

Reply:
It should. But there is a bug for CSI:  https://github.com/kubernetes-csi/external-provisioner/issues/195. If we simply delete a PVs, the CSI state will be in disorder. The consequence is the disk will not be removed in IssS provider.

This code will be refactored once this issue is fixed.